egs2/TEMPLATE/s2t1/s2t.sh

```diff
@@ -0,0 +1,1853 @@
+
+        # 1. Split the key file
+        key_file=${_data}/${_scp}
+        split_scps=""
+        if "${use_k2}"; then
+          # Now only _nj=1 is verified if using k2
+          _nj=1
+        else
+          _nj=$(min "${inference_nj}" "$(<${key_file} wc -l)")
+        fi
+
+        for n in $(seq "${_nj}"); do
+            split_scps+=" ${_logdir}/keys.${n}.scp"
+        done
+        # shellcheck disable=SC2086
+        utils/split_scp.pl "${key_file}" ${split_scps}
+
+        # 2. Submit decoding jobs
+        log "Decoding started... log: '${_logdir}/s2t_inference.*.log'"
+        rm -f "${_logdir}/*.log"
+        # shellcheck disable=SC2046,SC2086
+        ${_cmd} --gpu "${_ngpu}" JOB=1:"${_nj}" "${_logdir}"/s2t_inference.JOB.log \
+            ${python} -m espnet2.bin.s2t_inference${inference_bin_tag} \
+                --batch_size ${batch_size} \
+                --ngpu "${_ngpu}" \
+                --data_path_and_name_and_type "${_data}/${_scp},speech,${_type}" \
+                --key_file "${_logdir}"/keys.JOB.scp \
+                --s2t_train_config "${s2t_exp}"/config.yaml \
+                --s2t_model_file "${s2t_exp}"/"${inference_s2t_model}" \
+                --output_dir "${_logdir}"/output.JOB \
+                ${_opts} ${inference_args} || { cat $(grep -l -i error "${_logdir}"/s2t_inference.*.log) ; exit 1; }
+
+        # 3. Calculate and report RTF based on decoding logs
+        if [ -z ${inference_bin_tag} ]; then
+            log "Calculating RTF & latency... log: '${_logdir}/calculate_rtf.log'"
+            rm -f "${_logdir}"/calculate_rtf.log
+            _fs=$(python3 -c "import humanfriendly as h;print(h.parse_size('${fs}'))")
+            _sample_shift=$(python3 -c "print(1 / ${_fs} * 1000)") # in ms
+            ${_cmd} JOB=1 "${_logdir}"/calculate_rtf.log \
+                pyscripts/utils/calculate_rtf.py \
+                    --log-dir ${_logdir} \
+                    --log-name "s2t_inference" \
+                    --input-shift ${_sample_shift} \
+                    --start-times-marker "speech length" \
+                    --end-times-marker "best hypo" \
+                    --inf-num ${num_inf} || { cat "${_logdir}"/calculate_rtf.log; exit 1; }
+        fi
+
+        # 4. Concatenates the output files from each jobs
+        # shellcheck disable=SC2068
+        for ref_txt in ${ref_text_files[@]}; do
+            suffix=$(echo ${ref_txt} | sed 's/text//')
+            for f in token token_int score text; do
+                if [ -f "${_logdir}/output.1/1best_recog/${f}${suffix}" ]; then
+                    for i in $(seq "${_nj}"); do
+                        cat "${_logdir}/output.${i}/1best_recog/${f}${suffix}"
+                    done | sort -k1 >"${_dir}/${f}${suffix}"
+                fi
+            done
+        done
+
+    done
+fi
+
+
+if [ ${stage} -le 13 ] && [ ${stop_stage} -ge 13 ] && ! [[ " ${skip_stages} " =~ [[:space:]]13[[:space:]] ]]; then
+    log "Stage 13: Scoring"
```

Comment:
I guess you are still working on the scoring. Will check it later after it is settled~